src/doc/examples/lock-nodes.md

```diff
@@ -0,0 +1,184 @@
+# Examples
+
+## Node depended resources
+
+Create resources that are node depended is in many cases very helpful. That means a job must pick a target node where the requested resource is available.
+
+```groovy
+// allocate node
+node('some-build-node') {
+  // Lock resource named *whatever-resource-some-build-node*
+  lock("whatever-resource-${env.NODE_NAME}") {
+    echo "I am no node ${env.NODE_NAME} and locked resource ${env.LOCKED_RESOURCE}"
+  }
+}
+```
+
+But much more useful is lock node first.
+
+```groovy
+// Lock resource named *some-build-node*
+lock('some-build-node') {
+  // allocate node
+  node(env.LOCKED_RESOURCE) {
+
+    echo "I am no node ${env.NODE_NAME} and locked resource ${env.LOCKED_RESOURCE}"
+  }
+}
+```
+
+Let explain in more complex use case.
+
+*Request:*
+Our job tests server-client integration. That means we need 2 nodes (1 server and 1 client).
+On every node must be test sources up-to-date.
+Tests are running only on client side.
+
+*Solution:*
+Create 2 nodes:
+
+- node-server
+- node-client
+
+and execute it parallel like this:
+
+```groovy
+Map stages = [:];
+stages['server'] = {
+  node('node-server') {
+    prepareTests()
+    startServer()
+  }
+}
+stages['client'] = {
+  node('node-client') {
+    prepareTests()
+    startClientTest()
+  }
+}
+// execute all prepare stages synchronous
+parallel stages
+
+// Prepare tests on node
+void prepareTests() {
+  checkout([$class: 'GitSCM',
+            branches: [[name: 'master']]
+           ])
+}
+// Start server
+void startServer() {
+    echo 'Server will be started in few seconds'
+    sh 'mvn clean hpi:run'
+    echo 'Server is done'
+}
+// Start client
+void startClientTest() {
+  sleep 20
+  sh 'mvn clean verify'
+}
+```
+
+It looks pretty fine and easy, but !!!.
+
+Executing all steps parallel might leads to timing issues, because checkout on server-node might takes much longer then on client-node. This is serious issue because the client starts before the server and can not connect to server.
```

Comment:
```suggestion
Executing all steps in parallel will lead to timing issues.
Allocation of the server node might be delayed because a server node is not available.
Checkout on the server node might take much longer than on the client node.
This is serious issue because the client starts before the server and can not connect to server.
```